src/redux/books/books.js

```diff
@@ -13,12 +18,57 @@ export const removeBook = (payload) => ({
   payload,
 });
 
+export const postBook = (payload) => ({
+  type: POST_BOOK,
+  payload,
+});
+
+export const getBooks = async () => {
+  const response = await fetch(API_URL);
+  const bookItems = await response.json();
+  return bookItems;
+};
+
+export const getBookItemsFromApi = () => async (dispatch) => {
+  const bookItems = getBooks();
+  bookItems.then((bookItem) => {
+    Object.keys(bookItem).forEach((key) => {
+      const bookId = { id: key };
+      dispatch(addBook({ ...bookId, ...bookItem[key][0] }));
+    });
+  });
+};
+
 const reducer = (state = initialState, action) => {
   switch (action.type) {
     case ADD_BOOK:
       return [...state, action.payload];
     case REMOVE_BOOK:
+      fetch(`${API_URL}/${action.payload.id}`, {
+        method: 'delete',
+        body: JSON.stringify({
+          item_id: action.payload.id,
+        }),
+        headers: {
+          Accept: 'application/json',
+          'Content-Type': 'application/json',
+        },
+      });
```

Comment:
- [ ] I suggest the API functions should be created separately, that way the reducer would only return the state as expected. Kindly fix this.